dev_tools/profiling/quantum_volume.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tool to run the Quantum Volume benchmark defined by IBM in
+https://arxiv.org/abs/1811.12926."""
+
+import argparse
+from collections import defaultdict
+import math
+import sys
+from typing import DefaultDict, List
+
+import numpy as np
+
+import cirq
+
+
+def generate_model_circuit(num_qubits: int, depth: int) -> cirq.Circuit:
+    """Generates a model circuit with the given number of qubits and depth.
+
+    The generated circuit consists of `depth` layers of random qubit
+    permutations followed by random two-qubit gates that are sampled from the
+    Haar measure on SU(r).
+
+    Args:
+        num_qubits: The number of qubits in the generated circuit.
+        depth: The number of layers in the circuit.
+
+    Returns:
+      The generated circuit.
+    """
+    # Setup the circuit and its qubits.
+    qubits = [cirq.GridQubit(0, i) for i in range(num_qubits)]
+    circuit = cirq.Circuit()
+
+    # For each layer.
+    for _ in range(depth):
+        # Generate uniformly random permutation Pj of [0...n-1]
+        perm = np.random.permutation(num_qubits)
```

Comment:
This function should let you supply a random seed to replicability. The best way (IMO) of doing this is with the following pattern:

```python
def my_function(..., rs=None):
  if not isinstance(rs, np.random.RandomState):
    rs = np.random.RandomState(rs)
  ...
  perm = rs.permutation(num_qubits)
```

This lets you pass either (1) a numeric seed, (2) an existing `RandomState` so you can compose this function with other stochastic functions (3) `None`, which will seed according to dev/urandom

Reply:
Done, had to modify a few other files for full replicability.